packages/webview_flutter/webview_flutter_wkwebview/lib/src/ui_kit/ui_kit.dart

```diff
@@ -132,3 +141,44 @@ class UIView extends NSObject {
     );
   }
 }
+
+/// Responding to scroll view interactions.
+///
+/// Represent [UIScrollViewDelegate](https://developer.apple.com/documentation/uikit/uiscrollviewdelegate?language=objc).
+@immutable
+class UIScrollViewDelegate extends NSObject {
+  /// Constructs a [UIScrollViewDelegate].
+  UIScrollViewDelegate(
+      {this.scrollViewDidScroll,
+      super.observeValue,
```

Comment:
>...you didn't add support for it in Objective-C Flutter API

You mean [this](https://github.com/flutter/packages/pull/3444/files#diff-a06d9dace3b7d390131f9dd190504f6666e176d60fdec7dd36841029e5ddbb33R809)? 

That callback is being used so I don't think it can be removed. I was following the pattern of implementing `didFinishNavigation` of `WKNavigationDelegate`.

Reply:
I'm referring to the `super.observeValue` callback. Are you using this anywhere?

Reply:
> I'm referring to the `super.observeValue` callback. Are you using this anywhere?

Nope, I am gonna remove it.

If I am gonna use that callback I would need to implement the Objective-C Flutter API for it like [here](https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFObjectHostApi.m#L86), right? 

I am a littlle confuse, Is it correct that the `WKPreferences` is making the same mistake like mine (not implement the Objective-C Flutter API) [here](https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart#L248), or is it implemented somewhere else?

Reply:
> If I am gonna use that callback I would need to implement the Objective-C Flutter API for it like [here](https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFObjectHostApi.m#L86), right?

Correct

> I am a littlle confuse, Is it correct that the WKPreferences is making the same mistake like mine (not implement the Objective-C Flutter API) [here](https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart#L248), or is it implemented somewhere else?

`WKPreferences` only adds it to the `detached` constructor which shouldn't require it to be implemented based on when the constructor is used.